qiskit/algorithms/linear_solvers/exact_inverse.py

```diff
@@ -0,0 +1,44 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Exact inverse rotation."""
+
+from math import isclose
+import numpy as np
+from qiskit.circuit import QuantumCircuit, QuantumRegister
+from qiskit.extensions.quantum_initializer import UCRYGate
+
+
+class ExactInverse(QuantumCircuit):
+    r"""Exact inverse
+    .. math::
+
+        |x\rangle |0\rangle \mapsto \cos(1/x)|x\rangle|0\rangle + \sin(1/x)|x\rangle |1\rangle
+    """
+
+    def __init__(self, num_state_qubits: int, constant: float, name: str = '1/x') -> None:
```

Comment:
Can you add a docstring here for the constructor args. 

Is there a better name to give the parameter other than `constant`? Since you can give it any value it seems less like a constant in my book and it seems not at all descriptive for its role in the inverse, unlike num_state_qubits which seems more descriptive of what it does.

Reply:
Changed to `scaling` since what we compute is `sin(scaling/x)`.